src/sunmatrix/sparse/sunmatrix_sparse.c

```diff
@@ -847,22 +877,24 @@ int SUNMatScaleAdd_Sparse(sunrealtype c, SUNMatrix A, SUNMatrix B)
   /* access data arrays from A and B (return if failure) */
   Ap = Ai = Bp = Bi = NULL;
   Ax = Bx = NULL;
-  if (SM_INDEXPTRS_S(A)) { Ap = SM_INDEXPTRS_S(A); }
-  else { return (SUNMAT_MEM_FAIL); }
-  if (SM_INDEXVALS_S(A)) { Ai = SM_INDEXVALS_S(A); }
-  else { return (SUNMAT_MEM_FAIL); }
-  if (SM_DATA_S(A)) { Ax = SM_DATA_S(A); }
-  else { return (SUNMAT_MEM_FAIL); }
-  if (SM_INDEXPTRS_S(B)) { Bp = SM_INDEXPTRS_S(B); }
-  else { return (SUNMAT_MEM_FAIL); }
-  if (SM_INDEXVALS_S(B)) { Bi = SM_INDEXVALS_S(B); }
-  else { return (SUNMAT_MEM_FAIL); }
-  if (SM_DATA_S(B)) { Bx = SM_DATA_S(B); }
-  else { return (SUNMAT_MEM_FAIL); }
+  Ap      = SM_INDEXPTRS_S(A);
+  SUNCheck(Ap, SUN_ERR_ARG_CORRUPT);
+  Ai = SM_INDEXVALS_S(A);
+  SUNCheck(Ai, SUN_ERR_ARG_CORRUPT);
+  Ax = SM_DATA_S(A);
+  SUNCheck(Ax, SUN_ERR_ARG_CORRUPT);
+  Bp = SM_INDEXPTRS_S(B);
+  SUNCheck(Bp, SUN_ERR_ARG_CORRUPT);
+  Bi = SM_INDEXVALS_S(B);
+  SUNCheck(Bi, SUN_ERR_ARG_CORRUPT);
+  Bx = SM_DATA_S(B);
+  SUNCheck(Bx, SUN_ERR_ARG_CORRUPT);
 
   /* create work arrays for row indices and nonzero column values */
   w = (sunindextype*)malloc(M * sizeof(sunindextype));
+  SUNAssert(w, SUN_ERR_MALLOC_FAIL);
```

Comment:
So far I have yet to find a case where I have purposefully used `SUNCheck`. I think in this case, it was probably suggested by Dan because I had originally (several months ago) used `SUNCheck` for one of these values before we changed the meaning of `SUNAssert`.  

It may be that we end up not needing `SUNCheck` in addition to `SUNAssert`. For now, Ill change the lines above to `SUNAssert`.

Reply:
Coorection: I suppose the one case where `SUNCheck` has been used is for checking the result of a function that returns true/false rather than a `SUNErrCode`

Reply:
As part of error handling Phase III we can revisit the differences between `SUNCheck` and `SUNAssert` and determine if we should eliminate one or change the behavior of `SUNCheck` to something like a `SUNCheckAlways`. For now, I think its best to use `SUNAssert` everywhere unless its like the example above.